clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/provider/agent/TaskHealthCachingAgent.java

```diff
@@ -129,34 +129,38 @@ public TaskHealthCachingAgent(
             Keys.getTaskDefinitionKey(accountName, region, service.getTaskDefinition());
         TaskDefinition taskDefinition = taskDefinitionCacheClient.get(taskDefinitionCacheKey);
 
-        if (isContainerMissingNetworking(task)) {
+        if (taskDefinition == null) {
```

Comment:
Out of curiosity, why do we skip getting network bingings if taskDefinition does not exist in the cache? AFAICT NetworkBindings doesn't rely on the taskDefinition being present. Couldn't this lead to issues where taskDefinition hasn't yet been cached, and we skip the networkBindings/Interfaces check?

Reply:
hmm you know originally both methods were returning if `taskDefinition` was null, so it made sense to just do once upfront. but since it's actually NOT being used by `inferHealthNetworkBindedContainer` and we can't use the network mode to determine how to health check (which would require it sooner - see comment on line 150), you're right that we can just do that check once below. Will update!